packages/replay-next/src/suspense/createGenericCache.ts

```diff
@@ -24,6 +30,10 @@ export interface GenericCache<TExtraParams extends Array<any>, TParams extends A
   addValue(value: TValue, ...args: TParams): void;
   getStatus(...args: TParams): CacheRecordStatus | undefined;
   getCacheKey(...args: TParams): string;
+  subscribeToStatus: (
```

Comment:
:speech_balloon: I have to say that it _really_ feels like we're reinventing Redux at this point :)

Reply:
Redux manages data fetching and subscriptions too, so yeah for sure there are similarities.

This stuff is all being built on top of React core primitives though– and without any constraints of broader public use cases (or legacy usage patterns) so I think this is still an okay direction for us.